speechcommands/kwt.py

```diff
@@ -0,0 +1,219 @@
+from typing import Any
+import mlx.core as mx
+import mlx.nn as nn
+from mlx.utils import tree_flatten
+
+
+__all__ = ["KWT", "kwt1", "kwt2", "kwt3"]
+STD = 0.02
+
+
+class FeedForward(nn.Sequential):
+    def __init__(self, dim, hidden_dim, dropout=0.0):
+        super().__init__(
+            nn.Linear(dim, hidden_dim),
+            nn.GELU(),
+            nn.Dropout(dropout),
+            nn.Linear(hidden_dim, dim),
+            nn.Dropout(dropout),
+        )
+
+
+class Attention(nn.Module):
+    def __init__(self, dim, heads, dropout=0.0):
+        super().__init__()
+        self.heads = heads
+        self.scale = dim**-0.5
+        self.qkv = nn.Linear(dim, dim * 3, bias=False)
+        self.out = nn.Sequential(
+            nn.Linear(dim, dim), nn.Dropout(dropout)
+        )
+
+    def __call__(self, x):
+        b, n, _, h = *x.shape, self.heads
+        qkv = self.qkv(x)
+        qkv = qkv.reshape(b, n, 3, h, -1).transpose(2, 0, 3, 1, 4)
+        q, k, v = qkv
+        attn = (q @ k.transpose(0, 1, 3, 2)) * self.scale
+        attn = mx.softmax(attn, axis=-1)
+        x = (attn @ v).transpose(0, 2, 1, 3).reshape(b, n, -1)
+        x = self.out(x)
+        return x
+
+
+class Block(nn.Module):
+    def __init__(self, dim, heads, mlp_dim, dropout=0.0):
+        super().__init__()
+        self.attn = Attention(dim, heads, dropout=dropout)
+        self.norm1 = nn.LayerNorm(dim)
+        self.ff = FeedForward(dim, mlp_dim, dropout=dropout)
+        self.norm2 = nn.LayerNorm(dim)
+
+    def __call__(self, x):
+        x = self.attn(x)
+        x = self.norm1(x)
+        x = self.ff(x)
+        x = self.norm2(x)
```

Comment:
Is it intentional not to have the skip connections here?

Reply:
No, I missed it. I'll redo it and push a commit soon.

Reply:
I've fixed this in the latest commit, although changed README with results from only 10 epochs of training. I'll run full 100 epochs of training in a couple of days and add those results. For now can you just go through if there's anything else that you recommend changing?